csrc/mmdeploy/net/rknn/rknn_net.cpp

```diff
@@ -73,50 +128,75 @@ Result<void> RKNNNet::Init(const Value& args) {
   std::string content;
   OUTCOME_TRY(content, model.ReadFile(config.net));
   char* model_ptr = const_cast<char*>(content.data());
+#ifdef RK_MODELS
   int ret = rknn_init(&ctx_, model_ptr, content.size(), 0, NULL);
+#endif
+#ifdef RV_MODELS
+  int ret = rknn_init(&ctx_, model_ptr, content.size(), 0);
+#endif
   if (ret != RKNN_SUCC) {
-    MMDEPLOY_ERROR("Load .rknn failed! ret= {}", ret);
-    return Status(eInvalidArgument);
+    MMDEPLOY_ERROR("init rknn model with {} failed! ret: {}", config.net, ret);
+    return Status(eFail);
   }
 
   // Get Model Input Output Info
   rknn_input_output_num io_num;
   ret = rknn_query(ctx_, RKNN_QUERY_IN_OUT_NUM, &io_num, sizeof(io_num));
   if (ret != RKNN_SUCC) {
-    MMDEPLOY_INFO("model input num: {}, output num: {}\n", io_num.n_input, io_num.n_output);
-    MMDEPLOY_ERROR("rknn_query fail! ret= {}", ret);
+    MMDEPLOY_ERROR("rknn query 'RKNN_QUERY_IN_OUT_NUM' fail! ret: {}", ret);
     return Status(eFail);
   }
+  MMDEPLOY_DEBUG("model input num: {}, output num: {}", io_num.n_input, io_num.n_output);
+
+  auto get_tensor_shape = [](rknn_tensor_attr& attr) -> Result<TensorShape> {
+    TensorShape shape;
+    for (int i = 0; i < attr.n_dims; ++i) {
+      shape.push_back(attr.dims[i]);
+    }
+#ifdef RK_MODELS
+    //    return TensorDesc{device, data_type, shape, "#" + std::to_string(id)};
+    return shape;
+#endif
+#ifdef RV_MODELS
+    std::reverse(shape.begin(), shape.end());
+    return shape;
+#endif
+  };
 
   for (int i = 0; i < io_num.n_input; i++) {
-    rknn_tensor_attr input_attr;
-    input_attr.index = i;
-    ret = rknn_query(ctx_, RKNN_QUERY_INPUT_ATTR, &(input_attr), sizeof(rknn_tensor_attr));
+    rknn_tensor_attr attr;
+    attr.index = i;
+    ret = rknn_query(ctx_, RKNN_QUERY_INPUT_ATTR, &(attr), sizeof(rknn_tensor_attr));
     if (ret != RKNN_SUCC) {
-      MMDEPLOY_INFO("input tensors:\n");
-      dump_tensor_attr(&(input_attr));
-      MMDEPLOY_ERROR("rknn_query fail! ret= {}", ret);
+      MMDEPLOY_ERROR("rknn query 'RKNN_QUERY_INPUT_ATTR' fail! ret: {}", ret);
       return Status(eFail);
     }
-    input_attrs_.push_back(input_attr);
-    OUTCOME_TRY(auto data_type, GetMMDeployDataType(input_attr.type));
-    input_tensors_.emplace_back(TensorDesc{device_, data_type, {}, input_attr.name});
+    if (attr.type != RKNN_TENSOR_UINT8) {
+      MMDEPLOY_ERROR("MMDeploy SDK only supports RKNN-INT8 model");
+      return Status(eInvalidArgument);
+    }
+    input_attrs_.push_back(attr);
+    // Only support uint8 input data
+    OUTCOME_TRY(auto data_type, GetMMDeployDataType(RKNN_TENSOR_UINT8));
+    input_tensors_.emplace_back(
+        TensorDesc{device_, data_type, get_tensor_shape(attr).value(), "#" + std::to_string(i)});
   }
+  DebugRKNNTensorAttr("input", input_attrs_);
```

Comment:
seems it also works for release mode?

Reply:
It prints the rknn tensor attribute, working for both release and debug mode. I changed the name to `PrintRKNNTensorAttr` to clarify the purpose.